kyuubi-server/src/main/scala/org/apache/kyuubi/server/metadata/MetadataManager.scala

```diff
@@ -253,6 +256,31 @@ class MetadataManager extends AbstractService("MetadataManager") {
       TimeUnit.MILLISECONDS)
   }
 
+  @VisibleForTesting
+  private[metadata] def cleanupMetadata(maxAge: Long, batchSize: Int, batchInterval: Long): Unit = {
+    var needToCleanMetadata = true
+    var needToCleanKubernetesInfo = true
+
+    while (needToCleanMetadata || needToCleanKubernetesInfo) {
```

Comment:
let's have an enforced loop round threshold(maybe `100`, no need to be configurable， hardcode is fine) to guard this while loop.